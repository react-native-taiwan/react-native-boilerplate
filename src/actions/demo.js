import types from "../constants/ActionsType";

export const sub = payload => ({
  type: types.ADD,
  payload
});

export const less = payload => ({
  type: types.LESS,
  payload
});