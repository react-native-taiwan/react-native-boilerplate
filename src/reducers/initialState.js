import Immutable from "immutable";

const demo = Immutable.fromJS({
  count: 0,
});

export default {
  demo
};