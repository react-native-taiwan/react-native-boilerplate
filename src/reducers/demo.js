import types from '../constants/ActionsType';
import initialState from "./initialState";

export default function reducer(state = initialState.demo, { type }) {
  switch (type) {
    case types.ADD:
      return state.update("count", count => count + 1);
    case types.LESS:
    return state.update("count", count => count - 1);
    default:
      return state;
  }
}