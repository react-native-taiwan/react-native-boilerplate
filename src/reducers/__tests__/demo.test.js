import * as demoAction from "../../actions/demo";
import types from "../../constants/ActionsType";
import initialState from "../initialState";
import demoReducer from "../demo";

it("should 測試收到 加一action 後 reducer 的變化", () => {
  const state = demoReducer(initialState.demo, { type: "initial" });
  const payload = {count: 1};
  const action = demoAction.sub(payload);
  expect(state).toMatchSnapshot();
  expect(state.get("count")).toBe(0);
  const newState = demoReducer(state, { ...action });
  expect(newState).toMatchSnapshot();
  expect(newState.get("count")).toBe(1);;
});

it("should 測試收到 減一action 後 reducer 的變化", () => {
  const state = demoReducer(initialState.demo, { type: "initial" });
  const payload = {count: 1};
  const action = demoAction.less(payload);
  expect(state).toMatchSnapshot();
  expect(state.get("count")).toBe(0);
  const newState = demoReducer(state, { ...action });
  expect(newState).toMatchSnapshot();
  expect(newState.get("count")).toBe(-1);;
});

