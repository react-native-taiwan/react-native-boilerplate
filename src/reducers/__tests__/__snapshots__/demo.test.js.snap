// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should 測試收到 加一action 後 reducer 的變化 1`] = `
Immutable.Map {
  "count": 0,
}
`;

exports[`should 測試收到 加一action 後 reducer 的變化 2`] = `
Immutable.Map {
  "count": 1,
}
`;

exports[`should 測試收到 減一action 後 reducer 的變化 1`] = `
Immutable.Map {
  "count": 0,
}
`;

exports[`should 測試收到 減一action 後 reducer 的變化 2`] = `
Immutable.Map {
  "count": -1,
}
`;
